templates/eks/amazon-eks-nodepool-cf.yaml

```
@@ -736,6 +749,33 @@ Resources:
                       else
                         /etc/eks/bootstrap.sh ${ClusterName} --kubelet-extra-args '${KubeletExtraArguments}' ${BootstrapArguments}
                       fi
+
+                      # Note: manually applying stack tags onto root EBS
+                      # volumes, because automatic tag propagation for Amazon
+                      # EBS volumes created from block device mappings is not
+                      # supported as of 2021-03-11, source:
+                      # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-resource-tags.html
+                      # > All stack-level tags, including automatically created
+                      # > tags, are propagated to resources that CloudFormation
+                      # > supports. Currently, tags aren't propagated to Amazon
+                      # > EBS volumes that are created from block device
+                      # > mappings.
+                      if [ "${StackTags}" != "" ]; then
+                        STACK_TAG_ARGUMENTS=""
+                        for TAG in $(echo -n "${StackTags}" | tr "," "\n"); do
+                          TAG_KEY=$(echo -n "$TAG" | cut -d "=" -f 1)
+                          TAG_VALUE=$(echo -n "$TAG" | cut -d "=" -f 2)
+
+                          STACK_TAG_ARGUMENTS="$STACK_TAG_ARGUMENTS Key=$TAG_KEY,Value=$TAG_VALUE"
+                        done
+
+                        AWS_AVAIL_ZONE=$(curl http://169.254.169.254/latest/meta-data/placement/availability-zone)
+                        AWS_REGION="`echo \"$AWS_AVAIL_ZONE\" | sed 's/[a-z]$//'`"
```

```suggestion
                        AWS_REGION="$(echo "$AWS_AVAIL_ZONE" | sed 's/[a-z]$//')"
```
for consistent style